src/main/resources/com/github/jengelman/gradle/plugins/shadow/internal/unixStartScript.txt

```diff
@@ -156,7 +156,7 @@ fi
 
 # Escape application args
 save() {
-    for i do printf %s\\n "\$i" | sed "s/'/'\\\\''/g;1s/^/'/;\$s/\$/' \\\\/" ; done
+    for i do printf %s\\\\n "\\$i" | sed "s/'/'\\\\\\\\''/g;1s/^/'/;\\$s/\\$/' \\\\\\\\/" ; done
```

Comment:
For reference, gradle's script produces this line:
`for i do printf %s\\n "$i" | sed "s/'/'\\\\''/g;1s/^/'/;\$s/\$/' \\\\/" ; done`

So, besides for the `$i`, all of the escapes needed another escape.